libkbfs/folder_block_ops.go

```diff
@@ -2077,6 +2084,27 @@ func (fbo *folderBlockOps) startSyncWriteLocked(ctx context.Context,
 
 				fblock.IPtrs[i].BlockInfo = newInfo
 				md.AddRefBlock(newInfo)
+
+				// If this block is replacing a block from a previous,
+				// failed Sync, we need to take that block out of the
+				// refs list, and avoid unrefing it as well.
+				for i, ref := range si.op.RefBlocks {
```

Comment:
May be worth pulling this out into a function on `si`, say `removeReplacedBlock` or something?